lib/tapyrus/ext_key.rb

```diff
@@ -356,5 +360,9 @@ def self.support_version?(version)
       p = Tapyrus.chain_params
       [p.bip49_pubkey_p2wpkh_p2sh_version, p.bip84_pubkey_p2wpkh_version, p.extended_pubkey_version].include?(version)
     end
+
+    def master?
+      depth == 0
```

Comment:
In addition to `depth`, you also need to check number and fingerprint.

Reply:
Add check against number and fingerprint(97396b6)